cnd/src/network/comit_ln.rs

```diff
@@ -577,3 +577,281 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        ethereum::Address,
+        network::{derive_key_pair, transport, DialInformation, TokioExecutor},
+        swap_protocols::EthereumIdentity,
+    };
+    use anyhow::Context;
+    use bitcoin::secp256k1;
+    use futures::{pin_mut, task, StreamExt, TryFutureExt};
+    use libp2p::{
+        multihash::Sha3_256,
+        swarm::{SwarmBuilder, SwarmEvent},
+        Multiaddr, PeerId, Swarm,
+    };
+    use rand::thread_rng;
+
+    use crate::{asset::ethereum::FromWei, network::ComitNode};
+    use bitcoin::hashes::core::mem::swap;
+    use futures::{
+        future::{join3, join4},
+        join,
+        prelude::*,
+        ready,
+    };
+    use std::{str::FromStr, sync::Arc, thread::sleep, time};
+    use tokio::{macros::support::Pin, runtime, sync::Mutex};
+
+    fn random_swap_digest() -> SwapDigest {
+        SwapDigest::new(Sha3_256::digest(b"swapt digest"))
+    }
+
+    async fn start_communication(
+        alice_swarm: Arc<Mutex<Swarm<ComitLN>>>,
+        bob_swarm: Arc<Mutex<Swarm<ComitLN>>>,
+        alice_params: HanEtherereumHalightBitcoinCreateSwapParams,
+        bob_params: HanEtherereumHalightBitcoinCreateSwapParams,
+        alice_node_id: NodeLocalSwapId,
+        bob_node_id: NodeLocalSwapId,
+    ) {
+        println!("entered init...");
+
+        async {
+            // init bob first, because his swap has to be available for alice' announce
+            // message
+            sleep(time::Duration::from_secs(1));
```

Comment:
I do not think this will work as you expect, we should probably be using tokio `sleep` not `std::thread::sleep`.  ref: https://blog.hwc.io/posts/rust-futures-threadsleep-and-blocking-calls-inside-async-fn/.  Although from the code I cannot see why the sleeps are here in the first place :)